sound/soc/codecs/rt1308-sdw.c

```diff
@@ -265,9 +265,6 @@ static int rt1308_update_status(struct sdw_slave *slave,
 	/* Update the status */
 	rt1308->status = status;
 
-	if (status == SDW_SLAVE_UNATTACHED)
-		rt1308->hw_init = false;
```

Comment:
@shumingfan, I get that maybe for a 'SoftReset' due to a sync loss we don't need to reinitialize, but what does this have to do with resuming? On resume, you have to assume that the initializations are gone, we've cut the power to the codec.

Reply:
@plbossart I am not sure if HW engineer suggests the codec's power keeps or not. I also will check with the SW engineer who is in charge of SdW Windows driver next week.
I think we may use a register to check whether the codec's power disappeared ever.
To check this register, we decide whether to re-do the io_init function. How do you think?

Reply:
@shumingfan, this case was added specifically for the suspend-resume case, where definitively we will lose state.
I just don't see how removing this if() case will improve things?

Reply:
@plbossart Rander reported an issue to me that the speaker has no sound output after the resume.
The speaker should output immediately after the resume because he does the headphone/speaker switch test. Due to the unattached state happened, the io_init process and the power-up of the playback process will be interleaved. 
That's why I suggest the io_init function doesn't do again when the unattached state happened.